src/libs/ReportUtils.js

```diff
@@ -1952,6 +1953,44 @@ function shouldReportBeInOptionList(report, reportIDFromRoute, isInGSDMode, iouR
     return true;
 }
 
+/**
+ * Returns all the participants in the active report
+ *
+ * @param {Object} report The active report object
+ * @param {Object} personalDetails The personal details of the users
+ * @return {Array}
+ */
+function getAllParticipants(report, personalDetails) {
+    const participantAccountIDs = lodashGet(report, 'participantAccountIDs', []);
+
+    return _.chain(participantAccountIDs)
+        .map((accountID) => {
+            const userPersonalDetail = lodashGet(personalDetails, accountID, {displayName: personalDetails.displayName || Localize.translateLocal('common.hidden'), avatar: ''});
+            const userLogin = Str.isSMSLogin(userPersonalDetail.login || '') ? LocalePhoneNumber.formatPhoneNumber(userPersonalDetail.login) : userPersonalDetail.login;
+
+            return {
+                accountID: userPersonalDetail.accountID,
+                alternateText: userLogin || Localize.translateLocal('common.hidden'),
```

Comment:
@youssef-lr @luacmartins 
I did some code testing from my end as well. And here are the analysis
1) When we create the group we openReport API.
2) Open Report API response has merge command with `personalDetails` and `personalDetailsList` data. In both the data list we have list of users data. For me in both the list `I am getting personalDetails of the unknown users with their logins, names, etc`
3) This information is stored in onyx and ultimately comes to this line. 
4) Here I am always having personalDetails of the user we are searching. So `userLogin` is always present and it is not resolving to the hidden text for me.

So my assumption is for your accounts the API would not be returning `login` data or the `personalDetail` of the user itself. For me unless the API doesn't provide the data I don't think I am going to see the hidden text.

Note: Even for offline report creation, we set optimistic data with login for the openReport. Not sure how offline is working for you as well.

**My onyx data:**
<img width="733" alt="Screenshot 2023-06-21 at 10 53 44 PM" src="https://github.com/Expensify/App/assets/46707890/285d8ae6-1c2d-4eca-9cd3-cb5bd0286aab">


**OpenReport API response:**
https://github.com/Expensify/App/assets/46707890/3bc3c4e1-e9fa-41b4-a174-2a6efa62bf50

This is how i fetch this PR for testing

<img width="957" alt="Screenshot 2023-06-21 at 11 12 42 PM" src="https://github.com/Expensify/App/assets/46707890/61d7a70a-fcf2-4bb0-8982-1a36f2c6dec3">



Let me know if there is something I am doing wrongly. I am actually out of ideas 😵‍💫 .

Reply:
Let me know if there is something I am doing wrongly. I am actually out of ideas 😵‍💫

Reply:
Is the group you are creating with new users you haven't talked to before?

Reply:
yes. Always using users I haven't talked before.

Tried both combination
Combination 1:
1) Created brand new account 
2) created group with other of my accounts which this new account haven't interacted

Combination 2:
1) Created brand new account 
2) Created group with randoms emails

Both cases the same result. Even instead of new account in step 1, if i try a existing account and create group with new users, same result.

Reply:
Are you using production API or staging?

Reply:
That was gonna be my next question. These changes are only in the staging API at the moment.

Reply:
Oh it was the problem. By default local is hitting production, after toggling to "Use Staging Server" I am able to see this working.
🤕 🤕 🤕

Reply:
I wasn't aware of this API fix was only available in staging 😮‍💨